src/domain/swaps/contracts/decoders/gp-v2-decoder.helper.ts

```diff
@@ -1,18 +1,651 @@
+        name: 'tokens',
+        type: 'address[]',
+      },
+      {
+        components: [
+          {
+            internalType: 'uint256',
+            name: 'sellTokenIndex',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'buyTokenIndex',
+            type: 'uint256',
+          },
+          {
+            internalType: 'address',
+            name: 'receiver',
+            type: 'address',
+          },
+          {
+            internalType: 'uint256',
+            name: 'sellAmount',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'buyAmount',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint32',
+            name: 'validTo',
+            type: 'uint32',
+          },
+          {
+            internalType: 'bytes32',
+            name: 'appData',
+            type: 'bytes32',
+          },
+          {
+            internalType: 'uint256',
+            name: 'feeAmount',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'flags',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'executedAmount',
+            type: 'uint256',
+          },
+          {
+            internalType: 'bytes',
+            name: 'signature',
+            type: 'bytes',
+          },
+        ],
+        internalType: 'struct GPv2Trade.Data',
+        name: 'trade',
+        type: 'tuple',
+      },
+    ],
+    name: 'swap',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'vault',
+    outputs: [
+      {
+        internalType: 'contract IVault',
+        name: '',
+        type: 'address',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'vaultRelayer',
+    outputs: [
+      {
+        internalType: 'contract GPv2VaultRelayer',
+        name: '',
+        type: 'address',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    stateMutability: 'payable',
+    type: 'receive',
+  },
+] as const;
 
+export type GPv2OrderParameters = {
+  sellToken: `0x${string}`;
+  buyToken: `0x${string}`;
+  receiver: `0x${string}`;
+  sellAmount: bigint;
+  buyAmount: bigint;
+  validTo: number;
+  appData: `0x${string}`;
+  feeAmount: bigint;
+  kind: OrderKind;
+  partiallyFillable: boolean;
+  sellTokenBalance: SellTokenBalance;
+  buyTokenBalance: BuyTokenBalance;
+};
+
+/**
+ * Decoder for GPv2Settlement contract.
+ *
+ * The following is based on the CoW SDK implementation:
+ * @see https://github.com/cowprotocol/contracts/blob/1465e69f6935b3ef9ce45d4878e44f0335ef8531/src/ts/settlement.ts
+ */
 @Injectable()
-export class GPv2Decoder extends AbiDecoder<typeof abi> {
-  constructor(
```

Comment:
I've removed logging as I have a forthcoming refactor of the `AbiDecoder` that will simplify _all_ interactions with it (and include logging).